File: app/src/main/java/com/example/theduckcardchatsystem/viewmodel/SingleChatViewModel.java
package com.example.theduckcardchatsystem.viewmodel;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.recyclerview.widget.RecyclerView;

import com.example.theduckcardchatsystem.adapters.SingleChatAdapter;
import com.example.theduckcardchatsystem.repository.SingleChatRepository;

public class SingleChatViewModel extends AndroidViewModel {
    private Application application;
    private SingleChatRepository singleChatRepository;
    public SingleChatViewModel(@NonNull Application application) {
        super(application);
        this.application = application;
        singleChatRepository = new SingleChatRepository(application);
    }
    public void sendMessage(String message, String recevingUserId, String typeText){
     singleChatRepository.sendMessage(message, recevingUserId, typeText);

    }
    public void sendMessageAsIimage(String receivingUserID , String imageUrl,String messageKey){
        singleChatRepository.sendMessageAsImage(receivingUserID, imageUrl, messageKey);

    }
    public void saveToMainList(String id ,String type){
        singleChatRepository.saveToMainList(id,type);
    }
    public void initRecyclerView(String uid, SingleChatAdapter adapter, RecyclerView recyclerView){
        singleChatRepository.initRecyclerView(uid, adapter, recyclerView);
    }
    public void ClearChat(String uid){
        singleChatRepository.clearChat(uid);
    }
    public void DeleteChat(String uid){
        singleChatRepository.DeleteChat(uid);
    }
}

File: app/src/main/java/com/example/theduckcardchatsystem/viewmodel/SingleChatViewModel.kt
package com.example.theduckcardchatsystem.viewmodel

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import com.example.theduckcardchatsystem.repository.SettingsRepository
import androidx.lifecycle.LiveData
import com.example.theduckcardchatsystem.repository.SingleChatRepository
import com.example.theduckcardchatsystem.adapters.SingleChatAdapter
import androidx.recyclerview.widget.RecyclerView

class SingleChatViewModel(private val application: Application) : AndroidViewModel(
    application
) {
    private val singleChatRepository: SingleChatRepository
    fun sendMessage(message: String?, recevingUserId: String?, typeText: String?) {
        singleChatRepository.sendMessage(message, recevingUserId, typeText)
    }

    fun sendMessageAsIimage(receivingUserID: String?, imageUrl: String?, messageKey: String?) {
        singleChatRepository.sendMessageAsImage(receivingUserID, imageUrl, messageKey)
    }

    fun saveToMainList(id: String?, type: String?) {
        singleChatRepository.saveToMainList(id, type)
    }

    fun initRecyclerView(uid: String?, adapter: SingleChatAdapter?, recyclerView: RecyclerView?) {
        singleChatRepository.initRecyclerView(uid, adapter, recyclerView)
    }

    fun ClearChat(uid: String?) {
        singleChatRepository.clearChat(uid)
    }

    fun DeleteChat(uid: String?) {
        singleChatRepository.DeleteChat(uid)
    }

    init {
        singleChatRepository = SingleChatRepository(application)
    }
}